controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_types.go

```diff
@@ -223,6 +280,25 @@ type KubeadmControlPlaneStatus struct {
 	// Conditions defines current service state of the KubeadmControlPlane.
 	// +optional
 	Conditions clusterv1.Conditions `json:"conditions,omitempty"`
+
+	// LastRemediation stores info about last remediation performed.
+	// +optional
+	LastRemediation *LastRemediationStatus `json:"lastRemediation,omitempty"`
+}
+
+// LastRemediationStatus  stores info about last remediation performed.
+// NOTE: if for any reason information about last remediation are lost, RetryCount is going to restarts from 0 and thus
+// more remediation than expected might happen.
+type LastRemediationStatus struct {
+	// Machine is the machine name of the latest machine being remediated.
+	Machine string `json:"machine"`
+
+	// Timestamp is RFC 3339 date and time at which last remediation happened.
+	Timestamp metav1.Timestamp `json:"timestamp"`
```

Comment:
Wondering if we should use metav1.Time like in rolloutAfter and upstream in ObjectMeta

Reply:
done!